### PR TITLE
RS-645 - Use Glean telemetry for Focus data in unified_metrics view

### PR DESCRIPTION
**Pull request by akkomar**
This needs to be merged on April 3rd.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Comment by lucia-vargas-a** [comment]
For easier querying and as long as DS doesn't need to know from the view if the source is Legacy or Glean I'd merge both sources into a single app_name = 'Focus Android'. In that case the SELECT could be:

```
SELECT
    * EXCEPT(normalized_app_name) REPLACE (COALESCE(country, '??') AS country),
    IF(
      normalized_app_name IN ('Focus Android Glean', 'Focus Android'),
      'Focus Android',
      normalized_app_name
    ) AS normalized_app_name,
    `mozfun.norm.browser_version_info`(app_version) AS browser_version_info
```
Wdyt?

File: sql/moz-fx-data-shared-prod/telemetry/unified_metrics/view.sql [hunk 1]
```diff
@@ -7,5 +7,13 @@ SELECT
...
 WHERE
-  normalized_app_name != 'Focus Android Glean'
-  AND normalized_app_name != 'Focus Android Glean BrowserStack'
+  normalized_app_name NOT IN (
+    'Focus Android Glean',
+    'Focus Android Glean BrowserStack',
+    'Focus Android'
+  )
+  OR (
+    app_name IN ('Focus Android Glean', 'Focus Android Glean BrowserStack')
```
[the remark this answers is not in the document]
Comment (akkomar) [reply sql/moz-fx-data-shared-prod/telemetry/unified_metrics/view.sql:16]: Blah, copy-pasta, thank you for catching this!

**Review by lucia-vargas-a — COMMENTED** [review]
LGTM. I'd only merge of the Focus Android app_name, if DS is happy with that.

**Comment by akkomar** [comment]
> For easier querying and as long as DS doesn't need to know from the view if the source is Legacy or Glean I'd merge both sources into a single app_name = 'Focus Android'. In that case the SELECT could be:
> 
> ```
> SELECT
>     * EXCEPT(normalized_app_name) REPLACE (COALESCE(country, '??') AS country),
>     IF(
>       normalized_app_name IN ('Focus Android Glean', 'Focus Android'),
>       'Focus Android',
>       normalized_app_name
>     ) AS normalized_app_name,
>     `mozfun.norm.browser_version_info`(app_version) AS browser_version_info
> ```
> 
> Wdyt?

I think this makes sense.

BTW meaning of `normalized_app_name` seems to be slightly different across tables, compare mobile search aggregates: https://github.com/mozilla/bigquery-etl/blob/03beab4cb1028dec672892bf9b1b375b476e4c73/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L345-L346 ().

**Comment by akkomar** [comment]
> > For easier querying and as long as DS doesn't need to know from the view if the source is Legacy or Glean I'd merge both sources into a single app_name = 'Focus Android'. In that case the SELECT could be:
> > ```
> > SELECT
> >     * EXCEPT(normalized_app_name) REPLACE (COALESCE(country, '??') AS country),
> >     IF(
> >       normalized_app_name IN ('Focus Android Glean', 'Focus Android'),
> >       'Focus Android',
> >       normalized_app_name
> >     ) AS normalized_app_name,
> >     `mozfun.norm.browser_version_info`(app_version) AS browser_version_info
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Wdyt?
> 
> I think this makes sense.
> 
> BTW meaning of `normalized_app_name` seems to be slightly different across tables, compare mobile search aggregates:
> 
> https://github.com/mozilla/bigquery-etl/blob/03beab4cb1028dec672892bf9b1b375b476e4c73/sql/moz-fx-data-shared-prod/search_derived/mobile_search_clients_daily_v1/query.sql#L345-L346
> ().

@lucia-vargas-a I have implemented your suggestion. I think eventually this normalization should take place in `unified_metrics_v1` with an additional `app_name` field added, similar to `mobile_search_clients_daily_v1` mentioned above, however I don't have capacity to do that now.

**Pull request merged**